plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -49,8 +66,33 @@ export class KubernetesProxy {
     private readonly clusterSupplier: KubernetesClustersSupplier,
   ) {}
 
-  public createRequestHandler(): RequestHandler {
+  public createRequestHandler(
+    permissionApi: PermissionEvaluator,
+  ): RequestHandler {
     return async (req, res, next) => {
+      const token = getBearerTokenFromAuthorizationHeader(
+        req.header('authorization'),
+      );
+
+      const authorizeResponse = (
+        await permissionApi.authorize(
+          [
+            { permission: kubernetesProxyReadPermission },
+            { permission: kubernetesProxyCreatePermission },
+          ],
+          {
+            token,
+          },
+        )
+      )[0];
```

Comment:
What this effectively does is a batch of two authorization requests:

1. does the user with the ID token `token` have privileges for the `kubernetes.proxy.read` permission?
1. does the user with the ID token `token` have privileges for the `kubernetes.proxy.create` permission?

and then extracts whichever answer comes back first (the permission backend does not guarantee that it will return a batch of responses in the same order as the batch of requests)

I think we want to make a single [request](https://backstage.io/docs/reference/plugin-permission-common.authorizepermissionrequest) for a single permission here.

Reply:
I agree, the request has been changed to permission: kubernetesProxyPermission and makes only one auth request now.